File: mario-less/mario.c
#include <stdio.h>

int main(void)
{

    // Prompt the user for the pyramid's height
    int height;
    do
    {
        printf("Height: ");
        scanf("%i", &height);
    } while (height < 1 || height > 8); // Limit height between 1 and 8 for a standard CS50 requirement

    // Print the pyramid
    for (int i = 1; i <= height; i++)
    {
        // Print spaces for alignment
        for (int j = 0; j < height - i; j++)
        {
            printf(" ");
        }

        // Print hashes for the pyramid
        for (int k = 0; k < i; k++)
        {
            printf("#");
        }

        // Move to the next line after each row
        printf("\n");
    }
    // next

    return 0;
}

File: sentimental-hello/sentimental-mario-less/mario.py
# Prompt for height until it is between 1 and 8
while True:
    try:
        # Ask for input and convert to an integer
        num = int(input("Height: "))
        # Check if the number is within the desired range
        if 1 <= num <= 8:
            break
        else:
            print("Please enter a number between 1 and 8.")
    except ValueError:
        # Handle non-numeric input
        print("Invalid input. Please enter a number between 1 and 8.")

# Generate the right-aligned pyramid
for i in range(num):
    # Print spaces before hashes
    for j in range(num - i - 1):
        print(" ", end="")

    # Print hashes
    for j in range(i + 1):
        print("#", end="")

    # Move to the next line
    print()
